File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <bits/stdc++.h>
using namespace std;

int n, m, v, a, b, visited[1004];
vector<int> adj[1004];

void dfs(int here) {
  visited[here] = 1;
  cout << here << ' ';

  for (int there : adj[here]) {
    if (visited[there]) continue;
    dfs(there);
  }
  return;
}

void bfs(int here) {
  queue<int> q;
  visited[here] = 1;
  q.push(here);

  while (q.size()) {
    int here = q.front();
    q.pop();

    cout << here << ' ';

    for (int there : adj[here]) {
      if (visited[there]) continue;
      visited[there] = 1;
      q.push(there);
    }
  }
}

int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);

  cin >> n >> m >> v;

  for (int i = 0; i < m; i++) {
    cin >> a >> b;
    adj[a].push_back(b);
    adj[b].push_back(a);
  }
  for (int i = 1; i <= n; i++) {
    sort(adj[i].begin(), adj[i].end());
  }

  dfs(v);
  memset(visited, 0, sizeof(visited));
  cout << '\n';
  bfs(v);

  return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
// const input = fs.readFileSync('./input.txt').toString().trim().split('\n');

const [N, M, V] = input[0].split(' ').map((e) => +e);
const adj = Array.from({ length: N + 1 }, () => []);
const visited = Array.from({ length: N + 1 }, () => 0);

input.slice(1).map(e => e.split(' ').map(Number)).map((e) => {
  adj[e[0]].push(e[1])
  adj[e[1]].push(e[0])
});

adj.forEach(arr => arr.sort((a, b) => a - b));

const dfs = (here) => {
  visited[here] = 1;

  // 줄바꿈 없이 출력 
  process.stdout.write(here + ' ');

  for (let there of adj[here]){
    if(visited[there]) continue;
    dfs(there);
  }
  return;
}

const bfs = (start) => {
  visited[start] = 1;
  let q = [];
  q.push(start);

  while(q.length !== 0){
    const here = q.shift();
    process.stdout.write(here + ' ');
    
    for (let there of adj[here]){
      if(visited[there]) continue;

      visited[there] = 1;
      q.push(there);
    }
  }
  return;
}

const answer = () => {
  dfs(V);
  visited.fill(0);
  console.log('');
  bfs(V);
}

answer();